core/src/server/rpc_module.rs

```diff
@@ -142,6 +157,52 @@ impl CallOrSubscription {
 	}
 }
 
+/// The [`Weak`] equivalent of [`MethodCallback`]. This allows the [`Methods`]
+/// map to store aliases to the method callbacks and ensures that any dangling
+/// alias references are safely handled.
+///
+/// This type is an implementation detail of the method aliasing system and
+/// should not be used directly.
+#[derive(Clone, Debug)]
+pub enum WeakMethodCallback<'a> {
```

Comment:
dq: do we really need this Weak stuff if remove/merge actually takes of the subscription handlers?